packages/devtools_shared/lib/src/memory/memory_json.dart

```diff
@@ -17,17 +17,26 @@ class MemoryJson {
 
     // Iterate over all HeapSamples collected.
     data.map((f) {
-      if (result.isNotEmpty) result.write(',\n');
-      final encode = jsonEncode(f);
-      result.write('$encode');
+      final encode =
+          result.isNotEmpty ? encodeAnotherHeapSample(f) : encodeHeapSample(f);
+      result.write(encode);
     }).toList();
 
-    return '{"$_jsonPayloadField": {'
-        '"$_jsonVersionField": ${HeapSample.version}, "$_jsonDataField": [\n'
-        '$result'
-        '\n]\n}}';
+    return '$jsonHeader$result$jsonTrailer';
   }
 
+  static String get jsonHeader => '{"$_jsonPayloadField": {'
```

Comment:
while you are cleaning this up. Maybe switch this to use json encode instead of emitting json as strings.

Reply:
The reason I didn't was, for efficiency during devtools_server profile collecting.  Every HeapSample collected will then position the output file location to just before the end to footer part emitted so I can always write the HeapSample encoded and the footer then flush.

I was worried that if I didn't control the header and footer I'd need to parse the output to decide where the footer really was - this seemed simpler.

Reply:
makes sense